src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -236,6 +319,92 @@ public void onTraversalStart() {
     }
 
     /**
+     * Builds the exon collection given the values of user arguments.
+     *
+     * @throws UserException if there is some inconsistency in user arguments and inputs.
+     * @throws GATKException if there was any problem parsing the content of the exome file.
+     * @return never {@code null}.
+     */
+    private ExonCollection<? extends Locatable> resolveExonCollection() {
+        final ExonCollection<? extends Locatable> result;
+        if (exomeFile != null) {
+            result = resolveExonCollectionFromExomeFile();
+        } else if (hasIntervals()) {
+            final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+            if (sequenceDictionary == null) {
+                throw new UserException("missing reference sequence dictionary, as no input contains such a dictionary... please consider to provide fail-over dictionary using " +
+                        "a reference (e.g. --reference my_reference.fasta)");
+            }
+            result = ExonCollections.fromSimpleIntervalList(intervalArgumentCollection.getIntervals(sequenceDictionary));
+        } else {
+            throw new UserException(String.format("You must indicate the set of exon as input intervals (e.g. -L target-intervals.list) or a exome feature file (e.g. -%s my-targets.bed) ",EXOME_FILE_SHORT_NAME));
+        }
+        if (exonOutInfo.requiresUniqueExonName()) {
+            checkAllExonsHaveName(result);
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether all exons in the input exon collection have a designed name.
+     * @param result the query exon collection.
+     * @throws UserException if there is any exon in {@code result} without a name.
+     */
+    private void checkAllExonsHaveName(ExonCollection<? extends Locatable> result) {
+        if (result.exons().stream().anyMatch(e -> result.name(e) == null)) {
+            throw new UserException(String.format("Exon output info requested '%s' needs that each exon has a designated unique name/id but there is some with such a name: %s", exonOutInfo.name(),
+                    result.exons().stream().filter(e -> result.name(e) == null).map(e -> result.location(e).toString()).collect(Collectors.joining(", "))));
+        }
+    }
+
+    /**
+     * Constructs the exon collection from an exome-file passed by the user.
+     *
+     * @return never {@code null}.
+     */
+    private ExonCollection<? extends Locatable> resolveExonCollectionFromExomeFile() {
+        Utils.regularReadableUserFile(exomeFile,ExomeReadCounts.EXOME_FILE_SHORT_NAME);
+        final FeatureCodec<?, ?> codec = FeatureManager.getCodecForFile(exomeFile);
```

Comment:
@droazen Since FeatureCodec first argument extends Feature in the declaration is there a need to tell so here?